src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -61,10 +61,15 @@ abstract class Kernel implements KernelInterface, RebootableInterface, Terminabl
     protected $bundles = [];
 
     protected $container;
-    protected $environment;
     protected $debug;
     protected $booted = false;
     protected $startTime;
+    protected $mode;
```

Comment:
why making it protected ? This makes any BC layer harder (as child classes can do what they want with it). As there is a getter for it, child classes can still read it.